### PR TITLE
Compute transitively affected modules when no modules are directly affected

### PR DESCRIPTION
**Pull request by gsmet**
When a BOM property change affects managed dependencies but no module is directly affected (e.g. the property is only used in dependencyManagement), computeTransitivelyAffected was never called because the code returned early. This caused report mode to produce an empty affectedModules list, and trim/skip-tests modes to treat the build as unaffected.

This has been quite problematic for the Quarkus project as a change to our BOM doesn't properly trigger a build of the affected modules. We have a safeguard and we trigger the full build when no modules are affected but this is far from ideal.

If this makes sense, I would appreciate a quick release, thanks!

Assisted-By: Claude Opus 4.6 <noreply@anthropic.com>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved detection of modules affected by managed dependency and plugin version changes.
  * Fixed build set computation to properly include transitively affected modules in skip-tests and trim modes.

* **Tests**
  * Added test coverage for BOM property version change scenarios with transitive dependencies.

### CODE REVIEW

File: extension3/src/main/java/eu/maveniverse/maven/scalpel/extension3/internal/ScalpelLifecycleParticipant.java [hunk 1]
```diff
@@ -245,7 +245,28 @@ public void afterProjectsRead(MavenSession session) throws MavenExecutionExcepti
...
 
-            logger.info("Scalpel: {} modules directly affected: {}", directlyAffected.size(), keys(directlyAffected));
+            if (!directlyAffected.isEmpty()) {
+                logger.info(
+                        "Scalpel: {} modules directly affected: {}", directlyAffected.size(), keys(directlyAffected));
+            }
+
+            // Compute transitively affected modules (via changed managed deps/plugins)
+            Map<MavenProject, List<String>> transitivelyAffected = computeTransitivelyAffected(
+                    allProjects, directlyAffected, changedManagedDepGAs, changedManagedPluginGAs, session);
```
[the remark this answers is not in the document]
Comment (gsmet) [reply extension3/src/main/java/eu/maveniverse/maven/scalpel/extension3/internal/ScalpelLifecycleParticipant.java:255]: I think the issue is solved now. I force-pushed a fix.

**Comment by gsmet** [comment]
CodeRabbit review addressed!

**Review by gnodet — APPROVED** [review]
## In-Depth Review — PR #33

This is a real and impactful bug fix. The root cause is clear: `computeTransitivelyAffected` was never called when `directlyAffected` was empty, silently ignoring BOM-only property changes across all three modes (report, trim, skip-tests).

### Code Flow Change

The fix is structurally sound:
1. **First guard narrowed** with `&& changedManagedDepGAs.isEmpty() && changedManagedPluginGAs.isEmpty()` — only returns early when truly nothing changed
2. **`computeTransitivelyAffected` hoisted** above mode-specific branches — now called for all modes, not just report
3. **Second guard added** after transitive computation — catches the case where managed deps changed but no module actually uses them
4. **Report mode preserves categorization** by using `directlyAffected` (not `allAffected`) for `computeBuildSet`, setting `trimResult = null` when direct set is empty to avoid passing an empty set to `ReactorTrimmer`

### Correctness Assessment

| Mode | Verdict | Notes |
|------|---------|-------|
| Report | ✅ | `trimResult = null` handled correctly by `addTrimResultModules` and `addTransitivelyAffectedModules`. Transitive modules get `TRANSITIVE` category |
| Trim | ✅ | `allAffected` passed to `computeBuildSet` — transitive modules now correctly included in reactor |
| Skip-tests | ✅ | Same `allAffected` → transitive modules' tests will run as expected |

**Edge cases verified:**
- `forceBuildModules`: added to `directlyAffected` *before* the first guard → no interference
- No overlap between direct/transitive: `computeTransitivelyAffected` skips modules already in `directlyAffected`
- NPE safety: all collections properly initialized (`LinkedHashSet`/`LinkedHashMap`), `trimResult` null handled everywhere

### Test Assessment

Both new tests are well-constructed with realistic fixtures (BOM import scope, property-versioned managed deps, mock dependency resolution). `reportMode_bomPropertyChangeWithNoDirectlyAffectedModules` is particularly thorough — testing two managed dep GAs from one property, asserting `TRANSITIVE_DEPENDENCY` reason, `TRANSITIVE` category, and absence of unrelated modules.

**Minor gaps (non-blocking):**
- No trim-mode or skip-tests-mode test for the BOM-only scenario (the most impactful behavioral change)
- No test for the second guard (managed deps changed but no module uses them)

### Performance

For trim/skip-tests modes, `computeTransitivelyAffected` is now called when it wasn't before — meaning dependency resolution for all non-directly-affected modules. For a 600+ module reactor like Quarkus, this is mitigated by the `resolveCache` and partial resolution on failure. The overhead is justified by correctness.

### Cosmetic note

The early-return logic (write empty report / skip all tests / return) is now duplicated in two blocks. A small refactoring could extract a helper method, but this is purely cosmetic.

---

LGTM. Clean fix for a real problem, well-tested core scenario. 👍

**Review by cstamas — APPROVED** [review]
